src/interface/swarm.cpp

```diff
@@ -916,6 +918,7 @@ int Swarm::CountParticlesToSend_() {
     ApplyBoundaries_(total_noblock_particles, noblock_indices);
   }
 
+  // TODO(BRR) don't allocate dynamically
```

Comment:
This is meant as a future to do and not part of this PR, correct? I'm just asking because thisPR is about dynamic allocation.

Reply:
Yes there are still some outstanding dynamic allocation issues especially related to internal particle list sorting. One of the next things I want to work on is moving all this logic to device with `par_scan`, and that change should naturally clean up these dynamic alloc issues.